backend/tests/utils/getPort.test.ts

```diff
@@ -0,0 +1,53 @@
+import getPort from '../../src/utils/getPort';
+
+const originalEnv = process.env;
+
+beforeEach(() => {
+  jest.resetModules();
+  process.env = { ...originalEnv };
+});
+
+afterAll(() => {
+  process.env = originalEnv;
+});
+
+describe('getPort function', () => {
+  it('should throw an error if PORT is not set', async () => {
+    delete process.env.PORT;
+
+    await expect(getPort()).rejects.toThrow('Environment variable PORT must be set');
+  });
+
+  it('should return the port from environment when NODE_ENV is not "test"', async () => {
+    process.env.PORT = '3000';
+    process.env.NODE_ENV = 'development';
+
+    const port = await getPort();
+
+    expect(port).toBe(3000);
+  });
+
+  it('should increment the port by 1 if NODE_ENV is "test"', async () => {
```

Comment:
I think there's an extremely small chance that this test is non-deterministic because the test might run in a host that is using the 3001 port already